### PR TITLE
Add _logger.debug statements to cmodule default blas ldflags

### LINKED ISSUE #527: Add debug logging statements to cmodule blas
### Description

After the #444 breaking change, there have been many issues popping up about empty blas flags. It is very difficult to debug these remotely, so we should add `_logger.debug` statements to that part of the library to help us identify what is going on, why some flags are discarded and others are accepted.

### PR DESCRIPTION
**Pull request by lucianopaz**
closes #527

### CODE REVIEW

File: pytensor/link/c/cmodule.py [hunk 1]
```diff
@@ -2718,6 +2718,7 @@ def check_required_file(paths, required_regexs):
                     found = True
                     break
             if not found:
+                _logger.debug(f"Required file {req} not found")
```
Comment (michaelosthege) [inline pytensor/link/c/cmodule.py:2721]:
```suggestion
                _logger.debug("Required file '%s' not found", req)
```

File: pytensor/link/c/cmodule.py [hunk 2]
```diff
@@ -2779,9 +2780,14 @@ def check_libs(
         res = try_blas_flag(flags)
         if res:
             if any("mkl" in flag for flag in flags):
-                check_mkl_openmp()
+                try:
+                    check_mkl_openmp()
+                except Exception as e:
+                    _logger.debug(e)
+            _logger.debug(f"The following blas flags will be used {res}")
```
Comment (michaelosthege) [inline pytensor/link/c/cmodule.py:2787]:
```suggestion
            _logger.debug("The following blas flags will be used: '%s'", res)
```

File: pytensor/link/c/cmodule.py [hunk 3]
```diff
@@ -2779,9 +2780,14 @@ def check_libs(
...
             return res
         else:
+            _logger.debug(f"Supplied flags {res} failed to compile")
```
Comment (michaelosthege) [inline pytensor/link/c/cmodule.py:2790]:
```suggestion
            _logger.debug("Supplied flags '%s' failed to compile", res)
```

File: pytensor/link/c/cmodule.py [hunk 4]
```diff
@@ -2802,6 +2808,8 @@ def check_libs(
         # directory. We will include both in our searched library dirs
         searched_library_dirs.append(os.path.join(sys.prefix, "Library", "bin"))
         searched_library_dirs.append(os.path.join(sys.prefix, "Library", "lib"))
+    _logger.debug("Will search for BLAS libraries in the following directories:")
+    _logger.debug("{}".format("\n".join(searched_library_dirs)))
```
Comment (michaelosthege) [inline pytensor/link/c/cmodule.py:2812]:
```suggestion
    search_dirs = "\n".join(searched_library_dirs)
    _logger.debug("Will search for BLAS libraries in the following directories:\n%s", search_dirs)
```

**Review by michaelosthege — APPROVED** [review]
flake8 or so taught me to do it like this..

**Comment by lucianopaz** [comment]
> flake8 or so taught me to do it like this..

Do you really want to use old style string interpolation instead of the newer f-strings?

**Comment by michaelosthege** [comment]
> > flake8 or so taught me to do it like this..
> 
> Do you really want to use old style string interpolation instead of the newer f-strings?

It's because this way, if there's an error it gets logged properly

File: pytensor/link/c/cmodule.py (hunk 1 above)
Comment (twiecki) [reply pytensor/link/c/cmodule.py:2721]: Why?

**Comment by lucianopaz** [comment]
Out of curiosity, I looked a bit into why one would want to use old style string interpolation in logs. It seems like both formats will work fine and log properly, but the one you are suggested is only interpolated when the message is actually logged, while the modern interpolations are performed regardless of whether the message needs to be logged or not. I'll apply your suggestions and merge this.

**Comment by michaelosthege** [comment]
> Out of curiosity, I looked a bit into why one would want to use old style string interpolation in logs. It seems like both formats will work fine and log properly, but the one you are suggested is only interpolated when the message is actually logged, while the modern interpolations are performed regardless of whether the message needs to be logged or not. I'll apply your suggestions and merge this.

Thanks for the explanation!

I installed from this branch into my Docker image. The logger didn't show right away, and I had to set its level even though I had previously done a `logging.basicConfig(level=logging.DEBUG)`. No Idea why.

```
>>> pytensor.link.c.cmodule._logger.setLevel(logging.DEBUG)
>>> pytensor.link.c.cmodule._logger.debug("test")
DEBUG:pytensor.link.c.cmodule:test
>>> pytensor.link.c.cmodule.default_blas_ldflags()
DEBUG:pytensor.link.c.cmodule:Will search for BLAS libraries in the following directories:
/opt/mamba/lib/gcc/x86_64-conda-linux-gnu/12.3.0
/opt/mamba/lib/gcc
/opt/mamba/x86_64-conda-linux-gnu/lib
/opt/mamba/lib
/opt/mamba/x86_64-conda-linux-gnu/sysroot/lib64
/opt/mamba/x86_64-conda-linux-gnu/sysroot/usr/lib64
/opt/mamba/x86_64-conda-linux-gnu/lib
/opt/mamba/lib
/opt/mamba/x86_64-conda-linux-gnu/sysroot/lib64
/opt/mamba/x86_64-conda-linux-gnu/sysroot/usr/lib64
/opt/mamba/lib
DEBUG:pytensor.link.c.cmodule:Checking MKL flags with intel threading
DEBUG:pytensor.link.c.cmodule:Compiling for 64 bit architecture
DEBUG:pytensor.link.c.cmodule:Supplied flags  failed to compile
DEBUG:pytensor.link.c.cmodule:Supplied flags '' failed to compile
DEBUG:pytensor.link.c.cmodule:Supplied flags ['-lmkl_core', '-lmkl_rt', '-lmkl_intel_thread', '-liomp5', '-lpthread', '-Wl,-rpath,/opt/mamba/lib'] failed to compile
DEBUG:pytensor.link.c.cmodule:Checking MKL flags with GNU OpenMP threading
DEBUG:pytensor.link.c.cmodule:Compiling for 64 bit architecture
DEBUG:pytensor.link.c.cmodule:Supplied flags  failed to compile
DEBUG:pytensor.link.c.cmodule:Supplied flags '' failed to compile
DEBUG:pytensor.link.c.cmodule:Supplied flags ['-lmkl_core', '-lmkl_rt', '-lmkl_gnu_thread', '-lgomp', '-lpthread', '-Wl,-rpath,/opt/mamba/lib'] failed to compile
DEBUG:pytensor.link.c.cmodule:Checking Lapack + blas
DEBUG:pytensor.link.c.cmodule:Compiling for 64 bit architecture
DEBUG:pytensor.link.c.cmodule:Supplied flags  failed to compile
DEBUG:pytensor.link.c.cmodule:Supplied flags '' failed to compile
DEBUG:pytensor.link.c.cmodule:Supplied flags ['-llapack', '-lblas', '-lcblas', '-lm', '-Wl,-rpath,/opt/mamba/lib'] failed to compile
DEBUG:pytensor.link.c.cmodule:Checking blas alone
DEBUG:pytensor.link.c.cmodule:Compiling for 64 bit architecture
DEBUG:pytensor.link.c.cmodule:Supplied flags  failed to compile
DEBUG:pytensor.link.c.cmodule:Supplied flags '' failed to compile
DEBUG:pytensor.link.c.cmodule:Supplied flags ['-lblas', '-lcblas', '-Wl,-rpath,/opt/mamba/lib'] failed to compile
DEBUG:pytensor.link.c.cmodule:Checking openblas
DEBUG:pytensor.link.c.cmodule:Compiling for 64 bit architecture
DEBUG:pytensor.link.c.cmodule:Supplied flags  failed to compile
DEBUG:pytensor.link.c.cmodule:Supplied flags '' failed to compile
DEBUG:pytensor.link.c.cmodule:Supplied flags ['-lopenblas', '-lgfortran', '-lgomp', '-lm', '-fopenmp', '-Wl,-rpath,/opt/mamba/lib'] failed to compile
DEBUG:pytensor.link.c.cmodule:Failed to identify blas ldflags. Will leave them empty.
''
```

**Comment by michaelosthege** [comment]
With a bit more detailed logging:

```
DEBUG:pytensor.link.c.cmodule:Checking MKL flags with intel threading
DEBUG:pytensor.link.c.cmodule:Compiling for 64 bit architecture
DEBUG:pytensor.link.c.cmodule:Test compilation with flags '['-lmkl_core', '-lmkl_rt', '-lmkl_intel_thread', '-liomp5', '-lpthread', '-Wl,-rpath,/opt/mamba/lib', '-L/opt/mamba/lib']' resulted in
(False, False)
DEBUG:pytensor.link.c.cmodule:Supplied flags '['-lmkl_core', '-lmkl_rt', '-lmkl_intel_thread', '-liomp5', '-lpthread', '-Wl,-rpath,/opt/mamba/lib']' failed to compile
DEBUG:pytensor.link.c.cmodule:Supplied flags ['-lmkl_core', '-lmkl_rt', '-lmkl_intel_thread', '-liomp5', '-lpthread', '-Wl,-rpath,/opt/mamba/lib'] failed to compile
DEBUG:pytensor.link.c.cmodule:Checking MKL flags with GNU OpenMP threading
DEBUG:pytensor.link.c.cmodule:Compiling for 64 bit architecture
DEBUG:pytensor.link.c.cmodule:Test compilation with flags '['-lmkl_core', '-lmkl_rt', '-lmkl_gnu_thread', '-lgomp', '-lpthread', '-Wl,-rpath,/opt/mamba/lib', '-L/opt/mamba/lib']' resulted in
(False, False)
DEBUG:pytensor.link.c.cmodule:Supplied flags '['-lmkl_core', '-lmkl_rt', '-lmkl_gnu_thread', '-lgomp', '-lpthread', '-Wl,-rpath,/opt/mamba/lib']' failed to compile
DEBUG:pytensor.link.c.cmodule:Supplied flags ['-lmkl_core', '-lmkl_rt', '-lmkl_gnu_thread', '-lgomp', '-lpthread', '-Wl,-rpath,/opt/mamba/lib'] failed to compile
DEBUG:pytensor.link.c.cmodule:Checking Lapack + blas
DEBUG:pytensor.link.c.cmodule:Compiling for 64 bit architecture
DEBUG:pytensor.link.c.cmodule:Test compilation with flags '['-llapack', '-lblas', '-lcblas', '-lm', '-Wl,-rpath,/opt/mamba/lib', '-L/opt/mamba/lib']' resulted in
(False, False)
DEBUG:pytensor.link.c.cmodule:Supplied flags '['-llapack', '-lblas', '-lcblas', '-lm', '-Wl,-rpath,/opt/mamba/lib']' failed to compile
DEBUG:pytensor.link.c.cmodule:Supplied flags ['-llapack', '-lblas', '-lcblas', '-lm', '-Wl,-rpath,/opt/mamba/lib'] failed to compile
DEBUG:pytensor.link.c.cmodule:Checking blas alone
DEBUG:pytensor.link.c.cmodule:Compiling for 64 bit architecture
DEBUG:pytensor.link.c.cmodule:Test compilation with flags '['-lblas', '-lcblas', '-Wl,-rpath,/opt/mamba/lib', '-L/opt/mamba/lib']' resulted in
(False, False)
DEBUG:pytensor.link.c.cmodule:Supplied flags '['-lblas', '-lcblas', '-Wl,-rpath,/opt/mamba/lib']' failed to compile
DEBUG:pytensor.link.c.cmodule:Supplied flags ['-lblas', '-lcblas', '-Wl,-rpath,/opt/mamba/lib'] failed to compile
DEBUG:pytensor.link.c.cmodule:Checking openblas
DEBUG:pytensor.link.c.cmodule:Compiling for 64 bit architecture
DEBUG:pytensor.link.c.cmodule:Test compilation with flags '['-lopenblas', '-lgfortran', '-lgomp', '-lm', '-fopenmp', '-Wl,-rpath,/opt/mamba/lib', '-L/opt/mamba/lib']' resulted in
(False, False)
DEBUG:pytensor.link.c.cmodule:Supplied flags '['-lopenblas', '-lgfortran', '-lgomp', '-lm', '-fopenmp', '-Wl,-rpath,/opt/mamba/lib']' failed to compile
DEBUG:pytensor.link.c.cmodule:Supplied flags ['-lopenblas', '-lgfortran', '-lgomp', '-lm', '-fopenmp', '-Wl,-rpath,/opt/mamba/lib'] failed to compile
DEBUG:pytensor.link.c.cmodule:Failed to identify blas ldflags. Will leave them empty.
''
```

**Pull request merged**